### PR TITLE
Add support for parsing multiple FFL files for a given dataset

### PR DESCRIPTION
**Pull request by duncanmmacleod**
This PR addresses https://git.ligo.org/computing/helpdesk/-/issues/3841 by adding support for parsing multiple FFLs for a single dataset at Virgo.

cc @dethodav

### CODE REVIEW

**Review by rhyspoulton — COMMENTED** [review]
I have tested these changes and I can confirm that it is now able to deal with tag split across different ffl's with non-overlapping segments.

**Comment by rhyspoulton** [comment]
@duncanmmacleod is it possible to get this merged?

File: gwpy/io/ffldatafind.py [hunk 1]
```diff
@@ -147,15 +149,18 @@ def _ffl_path(site, tag, basedir=None):
...
-    with open(ffl, "r") as fobj:
-        return [
-            type(entry)(site, tag, entry.segment, entry.path)
-            for entry in _iter_cache(fobj, gpstype=float)
-        ]
+    entries = []
+    for ffl in _ffl_paths(site, tag, basedir=basedir):
+        with open(ffl, "r") as fobj:
+            entries.extend(
+                type(entry)(site, tag, entry.segment, entry.path)
```
Comment (asouthgate) [inline gwpy/io/ffldatafind.py:160]: why use type here instead of specific name?

**Review by asouthgate — APPROVED** [review]
Assuming ` type(entry)(site, tag, entry.segment, entry.path)` cannot be clearer, approved

File: gwpy/io/ffldatafind.py (hunk 1 above)
Comment (duncanmmacleod) [reply gwpy/io/ffldatafind.py:160]: No good reason other than using `type` saves from importing the actual type object and using it explicitly. However, that is probably not a very good reason...

Comment (duncanmmacleod) [reply gwpy/io/ffldatafind.py]: @asouthgate, I have updated this code to use the actual type object, which is clearer.

**Review by asouthgate — APPROVED** [review]

**Pull request merged**